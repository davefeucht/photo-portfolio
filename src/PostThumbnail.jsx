import React from "react";
import axios from "axios";
import PropTypes from "prop-types";

export default class PostThumbnail extends React.Component {
  constructor(props) {
    super(props);

    this.propTypes = {
      id: PropTypes.number,
      image: PropTypes.number,
      site: PropTypes.string,
      clickImage: PropTypes.func
    };

    this.state = {
      thumbnailUrl: ""
    };
  }

  _getPostThumbnail() {
    axios.get("http://" + this.props.site + "/wp-json/wp/v2/media/" + this.props.image + "/")
      .then(res => {
        let thumbnailImage = new Image();
        const thumbnailUrl = res.data.media_details.sizes.thumbnail.source_url;
        this.setState({ thumbnailUrl });
        thumbnailImage.src = thumbnailUrl;
      }, error => {
        const errorMsg = "Did not work: " + (error.response ? error.response : error);
        this.setState({ errorMsg });
      });
  } 

  _showFullPost(event) {
    event.preventDefault();
    this.props.clickImage(this.props.id);
  }

  componentWillMount() {
    this._getPostThumbnail();
  }

  render() {
    let divStyle = {backgroundImage: "url(" + this.state.thumbnailUrl + ")", backgroundRepeat: "no-repeat", backgroundPosition: "center center", backgroundSize: "auto 100%"};

    return(
      <div className="post-thumbnail" style={divStyle} onClick={this._showFullPost.bind(this)}>
      </div>
    );
  }
}