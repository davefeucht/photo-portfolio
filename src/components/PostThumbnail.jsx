import React from "react";
import axios from "axios";
import PropTypes from "prop-types";
import StyledPostThumbnail from "./styledComponents/StyledPostThumbnail.jsx";

export default class PostThumbnail extends React.Component {
  static propTypes = {
    id: PropTypes.number,
    image: PropTypes.number,
    site: PropTypes.string,
    clickImage: PropTypes.func
  };

  state = {
    thumbnailUrl: ""
  };

  _getPostThumbnail() {
    axios.get("http://" + this.props.site + "/wp-json/wp/v2/media/" + this.props.image + "/")
      .then(res => {
        let thumbnailImage = new Image();
        const thumbnailUrl = res.data.media_details.sizes.medium.source_url;
        this.setState({ thumbnailUrl });
        thumbnailImage.src = thumbnailUrl;
      }, error => {
        const errorMsg = "Did not work: " + (error.response ? error.response : error);
        this.setState({ errorMsg });
      });
  } 

  _showFullPost(event) {
    event.preventDefault();
    this.props.clickImage(this.props.id);
  }

  componentWillMount() {
    this._getPostThumbnail();
  }

  render() {
    const divStyle = {backgroundImage: "url(" + this.state.thumbnailUrl + ")", backgroundRepeat: "no-repeat", backgroundPosition: "center center", backgroundSize: "cover"};

    return(
      <StyledPostThumbnail style={divStyle} onClick={this._showFullPost.bind(this)} />
    );
  }
}