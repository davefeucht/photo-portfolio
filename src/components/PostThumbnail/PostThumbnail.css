div {
  height: 100%;
  cursor: pointer;
}