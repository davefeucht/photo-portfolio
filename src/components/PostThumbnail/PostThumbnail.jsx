/****************
* PostThumbnail component displays the thumbnail image for a Post
****************/

import React from "react";
import { runInAction } from "mobx";
import { observer } from 'mobx-react';
import './PostThumbnail.css';

const PostThumbnail = observer(({ stateStore, id, title, index }) => {
  const _showFullPost = () => {
    runInAction(() => {
      stateStore.setVisiblePost(id, title);
      stateStore.setShowAllPosts(false);
    });
  }

  const divStyle = {backgroundImage: "url(" + (stateStore.currentCategoryPosts[index].thumbnail_image ? stateStore.currentCategoryPosts[index].thumbnail_image : "") + ")", backgroundRepeat: "no-repeat", backgroundPosition: "center center", backgroundSize: "cover"};

  return(
    <div className="post-thumbnail" style={divStyle} onClick={_showFullPost.bind(this)}></div>
  );
});

PostThumbnail.displayName = 'PostThumbnail';

export default PostThumbnail;