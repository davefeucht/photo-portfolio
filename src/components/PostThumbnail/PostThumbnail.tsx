/** **************
* PostThumbnail component displays the thumbnail image for a Post
*************** */

import './PostThumbnail.css';

import { observer } from 'mobx-react';
import * as React from 'react';
import { Link, useLocation } from 'react-router-dom';

interface PostThumbnailProps {
    id: string,
    thumbnailImage: string
}

/*
 * currentCategoryPosts
*/

const PostThumbnail: React.FC<PostThumbnailProps> = ({ id, thumbnailImage = '' }) => {
    const divStyle = { backgroundImage: `url(${thumbnailImage})` };
    const location = useLocation();

    return (
        <Link
            to={{ pathname: `post/${id}` }}
            state={{ background: location }}
        >
            <div className="post-thumbnail" style={divStyle} />
        </Link>
    );
};

PostThumbnail.displayName = 'PostThumbnail';

export default observer(PostThumbnail);
