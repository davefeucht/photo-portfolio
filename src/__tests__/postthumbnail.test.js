import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import PostThumbnail from '../components/PostThumbnail/PostThumbnail';
import stateStore from '../StateStore/store';

const categoryPosts = [
    {
        thumbnail_image: 'some_url'
    }
];

const postId = 35;

test('PostThumbnail displays', () => {
  const store = new stateStore();
  let component;
  let tree;
  store.setCategoryPosts(categoryPosts);
  component = renderer.create(
      <MemoryRouter >
          <PostThumbnail stateStore={store} id={postId} index={0}></PostThumbnail>
      </MemoryRouter>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
