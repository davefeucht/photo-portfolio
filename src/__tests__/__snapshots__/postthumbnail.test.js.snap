// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostThumbnail displays 1`] = `
<a
  href="/post/35"
  onClick={[Function]}
>
  <div
    className="post-thumbnail"
    style={
      {
        "backgroundImage": "url(some_url)",
      }
    }
  />
</a>
`;
